soifunc/deband.py

```diff
@@ -21,30 +18,47 @@ def RetinexDeband(
     "medium" `threshold` in f3kdb is 48. I think that's a bit strong.
     16 might be a more sane starting point. Increase as needed.
 
-    `mask_threshold` determines how sensitive the mask is, lower values
-    should preserve more detail. It does not need to be manually scaled for bit-depth,
+    `mask_threshold` determines how sensitive the mask is.
+    Lower values should preserve more detail.
+    It does not need to be manually scaled for bit-depth,
     this function will do that automatically.
 
     This function does not add grain on its own. Use another function like
-    `kagefunc.adaptive_grain` to do that.
+    `vsdeband.sized_grain` to do that.
     """
+    assert clip.format
+    assert check_variable(clip, retinex_deband)
+
     if (
         clip.format.color_family != vs.YUV
         or clip.format.sample_type != vs.INTEGER
         or clip.format.bits_per_sample > 16
     ):
-        raise value_error("currently only supports 8-16 bit integer YUV input")
+        raise InvalidVideoFormatError(
+            retinex_deband, clip.format,
+            'The format {format.name} is not supported! It must be an 8-16bit integer YUV bit format!'
+        )
+
     bd_shift = 16 - clip.format.bits_per_sample
-    mask_threshold = mask_threshold >> bd_shift
+
     mask = (
-        kagefunc.retinex_edgemask(clip)
-        .std.Expr(f"x {mask_threshold} > x 0 ?")
-        .std.Inflate()
+        retinex(clip).std.Expr(f"x {mask_threshold >> bd_shift} > x 0 ?") .std.Inflate()
     )
+
     if showmask:
         return mask
-    threshold = threshold << 2
+
+    if 'y_2' not in core.neo_f3kdb.Deband.__signature__.parameters:  # type: ignore
```

Comment:
Wouldn't we want it to be `in` rather than `not in`? i.e. having `y_2` available assumes we're in the new neo_f3kdb?

Reply:
Yup, brainfart.